### DISCUSSION TITLE
Code style

### DISCUSSION THREAD

**Discussion by almirsarajcic** [3Stones-io/omedis#312] [General]
Let's use this thread to write down any code style conventions specific to this project, but also improvements we'd like to see.
I don't want to push my ideas immediately without discussing them first.

**Comment by almirsarajcic** [comment]
### Resource definitions order
In our codebase, every resource has its own order of definitions.
For example, in `Invitation`, `policies` come after `relationships`
https://github.com/wintermeyer/omedis/blob/1aa7760f77726f1a0f6369d1fb48bcd9f44f8232/lib/omedis/accounts/invitation.ex#L94-L131
but in `Activity`, `relationships` come at the end
https://github.com/wintermeyer/omedis/blob/1aa7760f77726f1a0f6369d1fb48bcd9f44f8232/lib/omedis/accounts/activity.ex#L250-L269

In other projects, I find having some kind of order (of any kind, but consistent across modules) helps with navigation. It might not be big of a deal, so I won't lobby for it too much.

Anyway, by default, Ash comes with the following `:spark` config which sets up the order of definitions in resources and domains.
https://github.com/optimumBA/ash_poc/blob/e6a9dfd3abdc04c484c34e3b35e0c63b2e791e89/ankra/config/config.exs#L15-L38

Together with the formatter config
https://github.com/optimumBA/ash_poc/blob/e6a9dfd3abdc04c484c34e3b35e0c63b2e791e89/ankra/.formatter.exs#L1-L14

it enables moving these definitions around automatically. With that approach, we (or the AI editor) can just add any new definitions to the end of the module, and the formatter will take care of the appropriate order.

**Reply by wintermeyer** [reply]
>  It might not be big of a deal, so I won't lobby for it too much.

I do lobby for it. It is such a time saver and helps onboarding new developers plus it helps existing developers everyday.

I am a big fan of the `:spark` config way. "Never send a human to do a machine's job."

https://www.youtube.com/watch?v=dJ4Bt2xtE9Q
